File: GolangSolutions/Array/M1732_找到最高海拔.go
/*
	@author Hyperspace
	@date 2022/11/19
	@file M1732_找到最高海拔.go
*/

package Array

func largestAltitude(gain []int) (ans int) {
	h := 0
	for _, v := range gain {
		h += v
		if h > ans {
			ans = h
		}
	}

	return
}

File: CPPSolutions/LeetCode/Array/M1732_找到最高海拔.cpp
#include <bits\stdc++.h>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int> &gain) {
        int n = gain.size();
        int ans = 0, h = 0;
        for (int i = 0; i < n; i++) {
            h += gain[i];
            if (h > ans) ans = h;
        }

        return ans;
    }
};